File: Readio/src/main/java/ninegle/Readio/library/mapper/LibraryMapper.java
package ninegle.Readio.library.mapper;

import java.util.List;

import org.springframework.data.domain.Page;

import ninegle.Readio.library.domain.Library;
import ninegle.Readio.library.dto.library.AllLibraryDto;
import ninegle.Readio.library.dto.library.LibraryListResponseDto;
import ninegle.Readio.library.dto.library.NewLibraryRequestDto;
import ninegle.Readio.library.dto.library.NewLibraryResponseDto;
import ninegle.Readio.library.dto.library.UpdateLibraryRequestDto;
import ninegle.Readio.library.dto.library.UpdateLibraryResponseDto;
import ninegle.Readio.user.domain.User;

public class LibraryMapper {

	//DTO에서 라이브러리로
	public static Library toNewLibraryEntity(NewLibraryRequestDto newLibraryRequestDto, User user) {
		return new Library(newLibraryRequestDto.libraryName(), user);
	}

	//라이브러리에서 DTO로
	public static NewLibraryResponseDto fromNewLibraryResponseDto(long libraryid, String libraryname, long userid) {
		NewLibraryResponseDto newLibraryResponseDto = NewLibraryResponseDto.builder()
			.libraryId(libraryid)
			.libraryName(libraryname)
			.userId(userid).build();
		return newLibraryResponseDto;
	}

	//전체 라이브러리 조회
	public static LibraryListResponseDto fromLibraryListResponseDto(Page<Library> librarypage) {
		List<AllLibraryDto> libraryList = librarypage.getContent().stream()
			.map(library -> AllLibraryDto.builder()
				.id(library.getId())
				.libraryName(library.getLibraryName())
				.createAt(library.getCreatedAt())
				.updateAt(library.getUpdatedAt())
				.build()).toList();
		return LibraryListResponseDto.builder()
			.allLibraries(libraryList)
			.totalCount(librarypage.getTotalElements())
			.page(librarypage.getNumber() + 1)
			.size(librarypage.getSize())
			.build();
	}

	//라이브러리 이름 수정
	public static String toLibraryName(UpdateLibraryRequestDto updateLibraryRequestDto) {
		return updateLibraryRequestDto.libraryName();
	}

	//라이브러리 이름 수정 후 Response
	public static UpdateLibraryResponseDto fromUpdateLibraryResponseDto(long libraryid, String libraryname) {
		return UpdateLibraryResponseDto.builder()
			.id(libraryid)
			.libraryName(libraryname).build();
	}
}

File: Readio/src/main/java/ninegle/Readio/library/mapper/LibraryMapper.kt
package ninegle.Readio.library.mapper

import ninegle.Readio.library.domain.Library
import ninegle.Readio.library.dto.library.*
import ninegle.Readio.user.domain.User
import org.springframework.data.domain.Page
import org.springframework.stereotype.Component

@Component
class LibraryMapper {
    //DTO에서 라이브러리로

	fun toNewLibraryEntity(newLibraryRequestDto: NewLibraryRequestDto, user: User): Library {
        return Library(newLibraryRequestDto.libraryName, user)
    }

    //라이브러리에서 DTO로
	fun fromNewLibraryResponseDto(libraryid: Long?, libraryname: String, userid: Long?): NewLibraryResponseDto {
        val newLibraryResponseDto = NewLibraryResponseDto(
            libraryId = libraryid,
            libraryName = libraryname,
            userId = userid)
        return newLibraryResponseDto
    }

    //전체 라이브러리 조회
    fun fromLibraryListResponseDto(librarypage: Page<Library>): LibraryListResponseDto {
        val libraryList: List<AllLibraryDto> = librarypage.content
            .map { library: Library ->
                AllLibraryDto(
                    id = library.id,
                    libraryName = library.libraryName,
                    createAt = library.createdAt,
                    updateAt = library.updatedAt)
            }
        return LibraryListResponseDto(
            allLibraries = libraryList,
            totalCount = librarypage.totalElements,
            page = librarypage.number+1,
            size = librarypage.size )

    }

    //라이브러리 이름 수정
	fun toLibraryName(updateLibraryRequestDto: UpdateLibraryRequestDto): String {
        return updateLibraryRequestDto.libraryName
    }

    //라이브러리 이름 수정 후 Response
	fun fromUpdateLibraryResponseDto(libraryid: Long?, libraryname: String): UpdateLibraryResponseDto {
        return UpdateLibraryResponseDto(
            id = libraryid,
            libraryName = libraryname)

    }
}
